app/web/readme.md

```diff
@@ -24,49 +81,59 @@ Communication with the backend is via [protobuf messages](https://github.com/pro
+docker run --rm -w /app -v $(pwd):/app registry.gitlab.com/couchers/grpc ./generate_protos.sh
+```
 
-- Go to the [CI pipelines](https://gitlab.com/couchers/couchers/-/pipelines/).
-- Search for the branch you want to generate the gRPC code from (usually `develop`).
-- Click the pipeline number.
-- Click the first pipeline step, "protos".
-- Click "download artifacts" on the right. This is a copy of the repo, but it has the generated gRPC code in it, so you can copy that from `couchers/app/web/src/proto` to your local clone of the repo.
+The TypeScript definitions will be generated into `app/web/proto` (all the other definitions will also be generated into the right places, for the Python backend, etc).
 
-Then, target the dev preview and API with the following command, instead of using `yarn start`, when running the app:
+You can always download the latest protos at <https://develop--protos.preview.couchershq.org/> for any of the languages we use if you don't want to set up Docker.
 
-```sh
-yarn cross-env NEXT_PUBLIC_API_BASE_URL=https://dev-api.couchershq.org yarn start
-```
+### Running against a local backend
 
-Alternatively, you can use `yarn start` if you update your local environment variables:
+The Quick Start instructions show you how to run only the frontend, pointing to the staging (next) backend/database. This is normally enough for most frontend development (since generally backend features are developed before the frontend implementation), but there may be cases where you want to run a modified backend. In order to do so, you need to do two things:
 
-- In `couchers/app/web/.env.development`, change `NEXT_PUBLIC_API_BASE_URL=http://localhost:8888` to `NEXT_PUBLIC_API_BASE_URL=https://dev-api.couchershq.org`
-- Remember not to commit this file to any pull requests!
+1. compile the protos locally (see above) and run the local backend, and
+2. update the frontend environment variables to point to the local backend.
 
-<details>
-<summary>Common problem: Getting logged out right after logging in</summary>
+The following shows the quick version. You'll need docker and docker compose installed.
 
-If you're getting logged out right after logging in, it's possible that 3rd party cookies are blocked in your browser. Since you're using localhost:3000, the cookie `couchers-sesh` coming from `https://dev-api.couchershq.org` is considered a 3rd party cookie.
+In one terminal, compile protos, run the backend and rest of the infrastructure:
 
-- Chrome allows to enable 3rd party cookies for specific websites in the cookie settings > Sites that can always use cookies. Enable "Including third-party cookies on this site"
-- Safari is all-or-nothing, in Preferences > Privacy > Prevent cross-site tracking. You have to disable it.
-</details>
+```sh
+## terminal 1
+cd app 
+# compile protos
+docker run --rm -w /app -v $(pwd):/app registry.gitlab.com/couchers/grpc ./generate_protos.sh
+# launch the rest of the docker containers
+docker compose up
```

Comment:
I think the first time you run it you'll also have to build the docker containers, or any time you change any env vars etc.

Reply:
`docker compose up` will build the ones it needs to build!